deploy/snapshot/internal/cuda/job.go

```
@@ -0,0 +1,181 @@
+
+func copyRegularFile(sourcePath, destinationPath string, destinationFlags int) error {
+	fd, err := unix.Open(sourcePath, unix.O_RDONLY|unix.O_CLOEXEC|unix.O_NOFOLLOW, 0)
+	if err != nil {
+		return err
+	}
+	source := os.NewFile(uintptr(fd), sourcePath)
+	if source == nil {
+		_ = unix.Close(fd)
+		return fmt.Errorf("open source %q", sourcePath)
+	}
+	defer source.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `source.Close` is not checked

(errcheck)